src/ts/localization/i18n.ts

```diff
@@ -0,0 +1,117 @@
+  language: 'en',
+  fallbackLanguages: ['en'],
+  translations: defaultTranslations,
+};
+//#endregion
+
+
+
+
+class I18n {
+  private language: string;
+  private vocabulary: BitmovinPlayerUiVocabulary;
+
+
+  constructor(config: BitmovinPlayerUiLocalizationConfig) {
+    this.setConfig(config);
+  }
+
+
+  public setConfig(config: BitmovinPlayerUiLocalizationConfig) {
+    const translations = { ...defaultTranslations, ...config.translations};
+    this.initializeLanguage(config.language, config.disableBrowserLanguageDetection, translations);
+    const fallbackLanguages = this.initializeFallbackLanguages(translations, config.fallbackLanguages);
+    this.initializeVocabulary(translations, fallbackLanguages);
+  }
+
+  private initializeLanguage(language: string, disableBrowserLanguageDetection: boolean, translations: TranslanslationsType ) {
+    const shouldDetectLanguage = !(disableBrowserLanguageDetection != null && disableBrowserLanguageDetection);
+
+    if (shouldDetectLanguage) {
+      let userLanguage = (window.navigator.language);
+      userLanguage = userLanguage.slice(0, 2);
+      // Check if  we also have it the language in the translations...
+      if (Object.keys(translations).findIndex((k) => k === userLanguage) !== -1) {
+        this.language = userLanguage;
+        return;
+      }
+    }
+
+    this.language = language;
+  }
+
+  private initializeFallbackLanguages(translations: TranslanslationsType, fallbackLanguages?: string[]) {
+    /**
+     * we extend fallback languages with user-defined translation keys.
+     * 'en' added statically to ensure it will be prioritized over the values of translation keys.
+     * new Set([]) will remove the duplicate and help us to add default fallbacks while respecting the order of user-defined fallbackLanguages
+     * removed 'translation' because it can't be fallback to itself.
+     */
+    return Array
+      .from(new Set([...(fallbackLanguages || []), 'en', ...Object.keys(translations)]))
+      .filter(l => l !== this.language);
+  }
+
+  private initializeVocabulary(translations: TranslanslationsType, fallbackLanguages: string[]) {
+    // reverse() to ensure we prioritize user-defined fallbackLanguages right after current language.
+    this.vocabulary = [...fallbackLanguages.reverse(), this.language] 
+      .reduce((vocab: BitmovinPlayerUiVocabulary, lang: string) => ({
+        ...vocab,
+        ...(translations[lang] || {}),
+    }), {});
+  }
+
+  private extractVariablesFromTranslationString(translation: string, values: {[key: string]: any} = {}) {
+    const translationVariables = Array.from(translation.match(/{([A-Z]|[a-z])+}/g) || []); // looks for {anyValue}
+
+    return translationVariables
+      .map(match => [match, match.slice(1, -1)])  // extract key for next step
+      .map(([match, key]) => ({ match, value: values[key]}));
+  }
+
+
+  public t(key: string, config: BitmovinPlayerUiTranslationConfig = {}): string {
```

Comment:
what's the upside of `t` instead of `translated`? as we get the translated word for the path